14-tidyr.Rmd

```diff
@@ -0,0 +1,154 @@
+str(gap_long)
+```
+
+Here we have used piping syntax which is similar to what we were doing in the previous lesson with dplyr. In fact, these are compatible and you can use a mix of tidyr and dplyr functions by piping them together
+
+Inside `gather()` we first name the new column for the new ID variable (`obs_type`), the name for the new amalgamated observation variable (`obs_value`), then the names of the old observation variable (`pop`,`lifeExp`,`gdpPercap`). Gather also allows the alternative syntax of using the `-` symbol to identify which variables are not to be gathered (i.e. ID variables)
+
+```{r}
+gap_long <- gapminder %>% gather(obs_type,obs_values,-continent,-country,-year)
+str(gap_long)
+```
+
+That may seem trivial with this particular dataframe, but sometimes you have 1 ID variable and 40 Observation variables. The flexibility is a huge time saver!
+
+> #### Challenge 1 {.challenge}
+>
+> Using `gap_long`, can you calculate the mean life expectancy, population, and gdpPercap for each continent?
+>**Hint:** use the `group_by()` and `summarize()` functions we learned in the dplyr lesson
+>
+
+#### From long to wide format with spread()
+Now just to double-check our work, lets use the opposite of `gather()` to spread our observation variables back out with the aptly named `spread()`
+
+```{r}
+gap_normal <- gap_long %>% spread(obs_type,obs_values)
+identical(gap_normal,gapminder)
+```
+
+That's great! We've gone from the longest format back to the intermediate and we didn't introduce any errors in our code.
+
+Now lets convert the long all the way to the wide. In the wide format, we will keep country and continent as ID variables and spread the observations across the 3 metrics (`pop`,`lifeExp`,`gdpPercap`) and time (`year`). First we need to create appropriate labels for all our new variables (time*metric combinations) and we also need to unify our ID variables to simplify the process of defining `gap_wide`
+
+```{r}
+gap_temp <- gap_long %>% unite(var_ID,continent,country,sep="_")
+str(gap_temp)
+
+gap_temp <- gap_long %>%
+    unite(ID_var,continent,country,sep="_") %>%
+    unite(var_names,obs_type,year,sep="_")
+str(gap_temp)
+```
+
+Using `unite()` we now have a single ID variable which is a combination of `continent`,`country`,and we have defined variable names. We're now ready to pipe in `spread()`  
+
+```{r}
+gap_wide <- gap_long %>% 
+    unite(ID_var,continent,country,sep="_") %>%
+    unite(var_names,obs_type,year,sep="_") %>%
+    spread(var_names,obs_values)
+str(gap_wide)
+```
+
+> #### Challenge 2 {.challenge}
+>
+> Can you take this 1 step further and create a `gap_ludicrously_wide` format data by spreading over countries, year and the 3 metrics?
+>**Hint** this new dataframe should only have 5 rows.
+>
+
+Now we have a great 'wide' format dataframe, but the `ID_var` could be more usable, let's separate them into 2 variables with `separate()`
```

Comment:
Pronoun disagreement. Change "them" to "it" or add "column" after `ID_var` and then change "them" to "it".